src/mds/mdstypes.cc

```diff
@@ -492,10 +492,15 @@ void feature_bitset_t::dump(Formatter *f) const {
 void feature_bitset_t::print(ostream& out) const
 {
   std::ios_base::fmtflags f(out.flags());
-  out << "0x";
-  for (int i = _vec.size() - 1; i >= 0; --i)
-    out << std::setfill('0') << std::setw(sizeof(block_type) * 2)
-        << std::hex << _vec[i];
+  int size = _vec.size();
+  if (!size) {
+    out << "0x0";
+  } else {
+    out << "0x";
```

Comment:
This can be made more succinct and therefore it'll be easier to read this -

```suggestion
  out << (size ? "0x" : "0x0")
```

Reply:
But we don't want the `for` loop to execute when the `size == 0`.